diff --git a/Makefile b/Makefile
index 6e34c94..d340fe9 100644
--- a/Makefile
+++ b/Makefile
@@ -821,11 +821,11 @@ KBUILD_CFLAGS	+= $(call cc-disable-warning, format-overflow)
 KBUILD_CFLAGS	+= $(call cc-disable-warning, address-of-packed-member)
 
 ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE
-KBUILD_CFLAGS += -O2
-KBUILD_RUSTFLAGS += -Copt-level=2
+KBUILD_CFLAGS += -O3 -march=native -fstack-protector-all -DFORTIFY_SOURCE=3 -fstack-protector-all -fstack-clash-protection -fcf-protection=full -mshstk
+KBUILD_RUSTFLAGS += -Copt-level=2 -Ctarget-cpu=native
 else ifdef CONFIG_CC_OPTIMIZE_FOR_SIZE
-KBUILD_CFLAGS += -Os
-KBUILD_RUSTFLAGS += -Copt-level=s
+KBUILD_CFLAGS += -Os -march=native -fstack-protector-all -DFORTIFY_SOURCE=3 -fstack-protector-all -fstack-clash-protection -fcf-protection=full -mshstk
+KBUILD_RUSTFLAGS += -Copt-level=s -Ctarget-cpu=native
 endif
